java/arcs/core/data/TypeUtils.kt

```diff
@@ -0,0 +1,140 @@
+package arcs.core.data
+
+import arcs.core.data.expression.Expression
+import arcs.core.data.expression.InferredType
+import arcs.core.data.expression.MapScope
+import arcs.core.data.expression.ParameterScope
+import arcs.core.data.expression.TypeEvaluator
+import arcs.core.type.Type
+
+private val primitiveToInferred = mapOf(
+  FieldType.Boolean to InferredType.Primitive.BooleanType,
+  FieldType.Long to InferredType.Primitive.LongType,
+  FieldType.Int to InferredType.Primitive.IntType,
+  FieldType.BigInt to InferredType.Primitive.BigIntType,
+  FieldType.Byte to InferredType.Primitive.ByteType,
+  FieldType.Short to InferredType.Primitive.ShortType,
+  FieldType.Char to InferredType.Primitive.ShortType,
+  FieldType.Float to InferredType.Primitive.FloatType,
+  FieldType.Double to InferredType.Primitive.DoubleType,
+  FieldType.Number to InferredType.Primitive.NumberType,
+  FieldType.Text to InferredType.Primitive.TextType,
+  FieldType.Instant to InferredType.Primitive.LongType)
```

Comment:
No, they are struct of 2 numbers (long, long). In JS they are (number, number). This is to model them as nanoseconds since epoch. But they're are modeled mathematically as BigInt. However in PaxelParser, times are modeled as milliseconds, not nanos. So this was done to make it consistent. (but it is not consistent with TS). I'll make a patch to PaxelParser to have it construct BigInt's for Instants and change the mapping.

Reply:
Ok, looking at DatabaseImpl, and other places in the code base both Kotlin and TS, it seems Instant.ofEpochMillis() is always used. That means, the nanosecond resolution is never used and we can model Instant as a raw long for now. However, to me, the Kotlin and TS code bases are broken. If we're going to claim we adhere to a spec, then we should support the whole spec. Which means nanosecond resolution.

Reply:
I think we should keep this as Long,  even in DatabaseImpl its *stored* as a Long.  We need to either decide if we are adhering to spec or not. If we are, we need to fix the whole stack implementation of Instant.